internal/safepath/safepath.go

```diff
@@ -36,15 +41,20 @@ func (s *SafePath) Close() error {
 
 // IsValid return true when path can still be used and wasn't cleaned up by Close.
 func (s *SafePath) IsValid() bool {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
 	return s.path != ""
 }
 
 // Path returns a safe, temporary path that can be used to access the original path.
 func (s *SafePath) Path() string {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
 	return s.path
```

Comment:
What happens if something tries to use this `SafePath` after close (because s.path is zeroed out currently).

Reply:
Theoretically:
1. If you use the result of `.Path()` obtained before path was `.Close()`d, that path is not guaranteed to point to the same thing as before `.Close()` was called or even exist at all. 
2. If you call `.Path()` after `.Close()` you get an empty path, and passing it anywhere should error with NotFound error. I guess we could add an explicit panic in this case?

Reply:
Panic seems like the thing to do here.

Reply:
Added a panic.